daemon/containerd/image_commit.go

```diff
@@ -312,16 +319,43 @@ func applyDiffLayer(ctx context.Context, name string, baseImg imagespec.DockerOC
 		}
 	}()
 
-	if _, err = differ.Apply(ctx, diffDesc, mount); err != nil {
+	if _, err = differ.Apply(ctx, diffDesc, mounts); err != nil {
 		return err
 	}
 
-	if err = sn.Commit(ctx, name, key); err != nil {
-		if cerrdefs.IsAlreadyExists(err) {
-			return nil
+	if !i.idMapping.Empty() {
```

Comment:
Perhaps good to add a comment inside this if-branch to describe what the intent is here (outline the issue).